lib/enterprise-event-stream.js

```diff
@@ -0,0 +1,252 @@
+ * @param {Function} callback - Called when done
+ * @returns {void}
+ */
+EnterpriseEventStream.prototype.writeStreamPositionToFile = function(file, callback) {
+	// We need to write both streamPosition and startDate, since streamPosition will be null until
+	// the first set of events is returned from Box.
+	fs.writeFile(file, JSON.stringify({
+		streamPosition: this._streamPosition,
+		startDate: this._options.startDate,
+		endDate: this._options.endDate,
+		eventTypeFilter: this._options.eventTypeFilter
+	}, null, 2), callback);
+};
+
+/**
+ * Fetch the next chunk of events
+ *
+ * If there are no events, poll until events are available.
+ * If an error occurs, emit the error but continuing polling as usual.
+ * @param {Function} callback - Passed the array of events
+ * @returns {void}
+ * @private
+ */
+EnterpriseEventStream.prototype.fetchEvents = function(callback) {
+	const self = this,
+		params = {
+			stream_type: 'admin_logs'
+		};
+
+	// Use the current stream position.
+	// Handle the case where the caller passes streamPosition === 0 instead of streamPosition === '0'.
+	if (this._streamPosition || this._streamPosition === 0) {
+		params.stream_position = this._streamPosition;
+	}
+
+	if (this._options.startDate) {
+		params.created_after = this._options.startDate;
+	}
+
+	if (this._options.endDate) {
+		params.created_before = this._options.endDate;
+	}
+
+	if (this._options.eventTypeFilter) {
+		params.event_type = this._options.eventTypeFilter;
+	}
+
+	if (this._options.chunkSize) {
+		params.limit = this._options.chunkSize;
+	}
+
+	this._client.events.get(params, function(err, result) {
+		if (err) {
+			self.emit('error', err);
+			// If there was a "permanent" error, we would call the callback with it here.
+			// But it's not clear which errors are truly permanent?
+			// If Box is down or returning errors for an extended period, we still want to resume when it recovers.
+			// So, continue polling at the regular frequency.
+			// Don't use a shorter retry interval (to avoid DDOSing Box).
+		} else if (result && result.entries) {
+			// console.log(`Fetched events from ${options.streamPosition || options.createdAfter} (limit ${options.limit}) => ${result.entries.length} events (next ${result.next_stream_position})`);
+
+			if (result.entries.length > 0) {
+				// Only update the stream position if there were events returned.
+				// The API currently returns next_stream_position = 0 if there are no events (may be a bug?).
+				// But we don't want to start over at the beginning in that case, so ignore it.
+				self._streamPosition = result.next_stream_position;
+
+				// Notify the reader of the new stream position.
+				// Stream readers can respond to the 'streamPosition' event to persist the stream position.
+				self.emit('streamPosition', self._streamPosition);
+
+				callback(null, result.entries);
+				return;
+			}
+		}
+
+		if (!self._options.pollingInterval) {
```

Comment:
How does this work if the user wanted all events from one specific day in the past, but those events didn't all fit into one page from the API?  Or if I want all the events from the start of today until now?  I think there's potential for a mismatch between how a user might think of "polling" and how it's being used here; I would expect that this applies only to "future" events, not that I'll only ever get one page.